jupyterhub/auth.py

```diff
@@ -123,17 +127,32 @@ class LocalAuthenticator(Authenticator):
         should I try to create the system user?
         """
     )
-    system_user_home = Unicode('', config=True,
-        help="""Specify root home directory for users if different from system default.
+    add_user_cmd = Command(config=True,
+        help="""add user command
         
-        Passed to `useradd -b`.
-        If specified, when system users are created their home directories will be created in
+        USERNAME will be replaced with the user's name.
+        The user's name will be used as the last argument, e.g.
         
-            system_user_home/username
+        You may want to consult the difference between adduser and useradd.
         
-        Only has an effect when users are created with `create_system_users=True`.
+        useradd -d /home/USERNAME -s /bin/bash
+        
+        will call
+        
+        useradd -d /home/river -s /bin/bash river
+        
+        when the user 'river' is created
         """
     )
+    def _add_user_cmd_default(self):
+        if sys.platform == 'darwin':
+            raise ValueError("I don't know how to create users on OS X")
+        elif which('pw'):
+            # Probably BSD
+            return ['pw', 'useradd', '-m']
+        else:
+            # This appears to be the Linux non-interactive adduser command:
+            return ['adduser', '-q', '--gecos', '', '--disable-password']
```

Comment:
This should be:

`['adduser', '-q', '--gecos', '""', '--disabled-password']`

* gecos has to be an empty string representation on the CLI `""`
* It is *--disabled-password*